Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/CloudFormationJsonWriter.cs

```diff
@@ -83,27 +83,49 @@ private void ProcessLambdaFunction(ILambdaFunctionSerializable lambdaFunction)
                 _jsonWriter.RemoveToken($"{propertiesPath}.Role");
             }
             
-            if (lambdaFunction.Attributes != null && lambdaFunction.Attributes.Any())
-                ProcessLambdaFunctionAttributes(lambdaFunction);
+            ProcessLambdaFunctionEvents(lambdaFunction);
         }
 
-        private void ProcessLambdaFunctionAttributes(ILambdaFunctionSerializable lambdaFunction)
+        private void ProcessLambdaFunctionEvents(ILambdaFunctionSerializable lambdaFunction)
         {
+            var currentSyncedEvents = new List<string>();
+            string eventName;
+            
             foreach (var attributeModel in lambdaFunction.Attributes)
             {
                 switch (attributeModel)
                 {
                     case AttributeModel<HttpApiAttribute> httpApiAttributeModel:
-                        ProcessHttpApiAttribute(lambdaFunction, httpApiAttributeModel.Data);
+                        eventName = ProcessHttpApiAttribute(lambdaFunction, httpApiAttributeModel.Data);
+                        currentSyncedEvents.Add(eventName);
                         break;
                     case AttributeModel<RestApiAttribute> restApiAttributeModel:
-                        ProcessRestApiAttribute(lambdaFunction, restApiAttributeModel.Data);
+                        eventName = ProcessRestApiAttribute(lambdaFunction, restApiAttributeModel.Data);
+                        currentSyncedEvents.Add(eventName);
                         break;
                 }
             }
+
+            var eventsPath = $"Resources.{lambdaFunction.Name}.Properties.Events";
+            var syncedEventsMetadataPath = $"Resources.{lambdaFunction.Name}.Metadata.SyncedEvents";
+            
+            if (!(_jsonWriter.GetToken(syncedEventsMetadataPath, new JArray()) is JArray previousSyncedEvents))
```

Comment:
This logic can be simplified with having only one return and one place of setting the syncevents json token.

```
if (_jsonWriter.GetToken(syncedEventsMetadataPath, new JArray()) is JArray previousSyncedEvents)
{
	foreach (var previousEventName in previousSyncedEvents.Select(x => x.ToObject<string>()))
	{
		if (!currentSyncedEvents.Contains(previousEventName))
			_jsonWriter.RemoveToken($"{eventsPath}.{previousEventName}");
	}
}


if (currentSyncedEvents.Any())
	_jsonWriter.SetToken(syncedEventsMetadataPath, new JArray(currentSyncedEvents));

```

Reply:
Fixed